com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/DeleteServicesStep.java

```diff
@@ -52,33 +52,49 @@ private void deleteServices(CloudControllerClient client, List<String> serviceNa
     }
 
     private void deleteService(CloudControllerClient client, String serviceName) {
+        CloudServiceInstance serviceInstance = null;
         try {
-            attemptToDeleteService(client, serviceName);
-        } catch (CloudOperationException e) {
-            switch (e.getStatusCode()) {
-                case NOT_FOUND:
-                    getStepLogger().warn(e, Messages.COULD_NOT_DELETE_SERVICE, serviceName);
-                    break;
-                case BAD_GATEWAY:
-                    throw new CloudServiceBrokerException(e);
-                default:
-                    throw e;
+            serviceInstance = client.getServiceInstance(serviceName);
+            List<CloudServiceBinding> bindings = serviceInstance.getBindings();
+            if (!CommonUtil.isNullOrEmpty(bindings)) {
+                logBindings(bindings);
+                getStepLogger().info(Messages.SERVICE_HAS_BINDINGS_AND_CANNOT_BE_DELETED, serviceName);
+                return;
             }
+
+            getStepLogger().info(Messages.DELETING_SERVICE, serviceName);
+            client.deleteService(serviceName);
+            getStepLogger().debug(Messages.SERVICE_DELETED, serviceName);
+        } catch (CloudOperationException e) {
+            processOperationException(e, serviceInstance, serviceName);
+        } catch (CloudException e) {
+            processAndWrapException(e, serviceInstance, serviceName);
         }
     }
 
-    private void attemptToDeleteService(CloudControllerClient client, String serviceName) {
-        CloudServiceInstance serviceInstance = client.getServiceInstance(serviceName);
-        List<CloudServiceBinding> bindings = serviceInstance.getBindings();
-        if (!CommonUtil.isNullOrEmpty(bindings)) {
-            logBindings(bindings);
-            getStepLogger().info(Messages.SERVICE_HAS_BINDINGS_AND_CANNOT_BE_DELETED, serviceName);
+    private void processOperationException(CloudOperationException e, CloudServiceInstance serviceInstance, String serviceName) {
+        if (e.getStatusCode() == HttpStatus.NOT_FOUND) {
+            getStepLogger().warn(e, Messages.COULD_NOT_DELETE_SERVICE, serviceName);
             return;
         }
+        if (e.getStatusCode() == HttpStatus.BAD_GATEWAY) {
+            e = new CloudServiceBrokerException(e);
+        } else {
+            e = new CloudControllerException(e);
+        }

```

Comment:
From line 76-84, you can extract in separate method in order to make the code more readable.